### PR TITLE
feat/va-carousel

### LINKED ISSUE #45: Carousel component
# Docs

- [ ] Do after [docs boilerplate](#17) is done.

### Props
- [x] `value: Number` - default: 0 - index of active slide
- [x] `autoplay: Boolean` - whether to scroll automatically
- [x] `autoplay-interval (*): Number` - default: 3000 - interval of the auto loop, in milliseconds
- [x] `loop: Boolean` - display the items in loop
- [ ] `height: String` - default: `auto` - the height of the carousel
- [x] `dots: Boolean` - show navigation dots
- [x] `arrows: Boolean` - show navigation arrow buttons
- [ ] `trigger (*): String` - default: 'click' - the way to activate the indicator. Optional value: click, hover.
- [ ] `effect: String` - default: `fade` - transition effect
- [ ] `vertical (*): Boolean` - whether to use a vertical display

### Slots
- [x] `default`
- [x] `control` - slot specific for carousel control

### Events
- [x] `input` - emitted when component's model changes

### Methods
- [x] `goTo` - go to specific panel
- [x] `next` - go to next panel
- [x] `prev` - go to previous panel

### Props
- [x] `name: String` - name of the item, can be used in goTo
- [x] `src: String`
- [x] #921

### Slots
- [x] `default`

# Design
- [ ] Present to @xx13 after finishing.

# Refs
* [Vuetify](https://vuetifyjs.com/en/components/carousels)
* [Quasar](https://quasar.dev/vue-components/carousel)
* [Ant Design](https://vue.ant.design/components/carousel/)
* [iView](http://iview.talkingdata.com/#/components/carousel-en)
* [Element UI](https://element.eleme.io/#/en-US/component/carousel)

### PR DESCRIPTION
**Pull request by kushich**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/docs/components/page-configs/va-carousel/api-options.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { ManualApiOptions } from 'vuestic-ui/src/services/api-docs/ManualApiOptions'
```
Comment (asvae) [inline packages/docs/components/page-configs/va-carousel/api-options.ts:1]: Needs to be more visible: 
image

Comment (asvae) [inline packages/docs/components/page-configs/va-carousel/api-options.ts:1]: draggable is kinda easy to break

File: packages/ui/src/components/vuestic-components/va-carousel/VaCarousel.vue [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+  @Watch('currentPage')
+  onCurrentPageChanged (val: number) {
+    this.$emit('input', val)
+  }
+
+  get innerStyles () {
+    return {
+      transform: `translate(${this.currentOffset}px, 0)`,
+      transition: this.dragging ? 'none' : '0.5s ease-out transform',
+      'ms-flex-preferred-size': `${this.c_width}px`,
```
Comment (asvae) [inline packages/ui/src/components/vuestic-components/va-carousel/VaCarousel.vue:110]: you can use camelCase here

File: packages/ui/src/components/vuestic-components/va-carousel/VaCarousel.vue [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+  },
+})
+
+@Component({
+  name: 'VaCarousel',
+  components: {
+    VaCarouselNavigation,
+    VaCarouselPagination,
+  },
+  provide () {
```
Comment (asvae) [inline packages/ui/src/components/vuestic-components/va-carousel/VaCarousel.vue:71]: Can you check this works in nuxt?

File: packages/docs/components/page-configs/va-carousel/page-config.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
+import { BlockType, ApiDocsBlock } from '../../../types/configTypes'
+import VaCarousel from 'vuestic-ui/src/components/vuestic-components/va-carousel/VaCarousel.vue'
+import apiOptions from './api-options'
+
+export default [
+  {
```
Comment (asvae) [inline packages/docs/components/page-configs/va-carousel/page-config.ts:6]: We probably can use helpers here, given some other changes are needed.

**Review by asvae — CHANGES_REQUESTED** [review]
Overall looks good, but some small tweaks are needed.

**Comment by kushich** [comment]
Outdated OR. It's easier to create a new solution

**Pull request closed without merging**